gpkit/constraints/geometric_program.py

```diff
@@ -44,35 +46,28 @@ class GeometricProgram(NomialData):
     """
 
     def __init__(self, cost, constraints, substitutions=None, verbosity=1):
-        # pylint: disable=too-many-locals,super-init-not-called
-        # note: any substitutions in constraints are ignored -- overwritten
-        # by substitutions kwarg. This will be fixed when GeometricProgram
-        # inherits from ConstraintSet.
-        self.cost = cost
-        self.constraints = constraints
-        self.substitutions = substitutions if substitutions else {}
-
-        ## Create list of substituted posynomials <= 1
+        # pylint:disable=super-init-not-called
+        # initialize attributes modified by internal methods
+        self.result = None
+        self.solver_log = None
+        self.solver_out = None
+        CostedConstraintSet.__init__(self, cost, constraints)
+        # NOTE: unnecessary work done above in creating self.substitutions
+        self.substitutions = substitutions
```

Comment:
great inline comment, but I'm slightly confused: it doesn't make sense to pass `substitutions` to `CostedConstraintSet.__init__` and have that manage all the subs?